Various test failures on docker golang image

**Describe the bug**
`go test` does not succeeds out of the box.

**To Reproduce**
I use the golang docker container to run the tests.
```
$ docker run -it --rm golang
> go version
go version go1.14.4 linux/amd64
> go get -t github.com/shirou/gopsutil/...
> go test github.com/shirou/gopsutil/...
```

**Expected behavior**
I expect the tests to succeed, however that is not the case.

```
--- FAIL: TestUsers (0.00s)
    host_test.go:70: Users is empty
FAIL
FAIL	github.com/shirou/gopsutil/host	0.045s
ok  	github.com/shirou/gopsutil/internal/common	0.003s
ok  	github.com/shirou/gopsutil/load	0.022s
ok  	github.com/shirou/gopsutil/mem	0.005s
--- FAIL: TestGetProcInodesAll (0.00s)
    net_linux_test.go:87: 
        	Error Trace:	net_linux_test.go:87
        	Error:      	Should NOT be empty, but was map[]
        	Test:       	TestGetProcInodesAll
FAIL
FAIL	github.com/shirou/gopsutil/net	0.011s
--- FAIL: Test_Process_memory_maps (0.00s)
    process_test.go:116: memory map get error {"path":"","rss":0,"size":0,"pss":0,"sharedClean":0,"sharedDirty":0,"privateClean":0,"privateDirty":0,"referenced":0,"anonymous":0,"swap":0}
... THE LINE ABOVE MANY TIMES ...
FAIL
FAIL	github.com/shirou/gopsutil/process	1.919s
FAIL
```

**Environment (please complete the following information):**
 - [ ] Linux: [paste contents of `/etc/os-release` and the result of `uname -a`]

```
# uname -a
Linux 343ae37ea285 5.7.0-1-amd64 #1 SMP Debian 5.7.6-1 (2020-06-24) x86_64 GNU/Linux
# cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

**Additional context**
I work on the packaging of gopsutil for Debian.

Note that Debian already has a patch to skip tests for the 1st test failure, ie `TestUsers`. I don't know why it fails, either this test requires root, either it's expected to fail in a container / chroot. In any case, we skip it when we build the Debian package.

The second test, `TestGetProcInodesAll`, is skipped if I set `CIRCLECI=true` in the environment, so I'm not worried about this failure, I just assume that we lack some capability in the docker container.

The 3rd test however, `Test_Process_memory_maps`, is a new failure that I've never seen before.

If you have any hint regarding why this test fail, maybe if it's expected to fail for non-privileged user, or if running in a container or chroot, that would be helpful.

Thanks!

Reference: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=964045

--- a/host/host_test.go
+++ b/host/host_test.go
@@ -67,7 +67,7 @@ func TestUsers(t *testing.T) {
 	}
 	empty := UserStat{}
 	if len(v) == 0 {
-		t.Fatal("Users is empty")
+		t.Skip("Users is empty")
 	}
 	for _, u := range v {
 		if u == empty {

--- a/net/net_linux_test.go
+++ b/net/net_linux_test.go
@@ -3,6 +3,7 @@ package net
 import (
 	"fmt"
 	"io/ioutil"
+	"net"
 	"os"
 	"strings"
 	"syscall"
@@ -77,9 +78,27 @@ func TestIOCountersByFileParsing(t *testing.T) {
 }
 
 func TestGetProcInodesAll(t *testing.T) {
-	if os.Getenv("CIRCLECI") == "true" {
-		t.Skip("Skip CI")
-	}
+	waitForServer := make(chan bool)
+	go func() { // TCP listening goroutine to have some opened inodes even in CI
+		addr, err := net.ResolveTCPAddr("tcp", "localhost:0") // dynamically get a random open port from OS
+		if err != nil {
+			t.Skip("unable to resolve localhost:", err)
+		}
+		l, err := net.ListenTCP(addr.Network(), addr)
+		if err != nil {
+			t.Skip(fmt.Sprintf("unable to listen on %v: %v", addr, err))
+		}
+		defer l.Close()
+		waitForServer <- true
+		for {
+			conn, err := l.Accept()
+			if err != nil {
+				t.Skip("unable to accept connection:", err)
+			}
+			defer conn.Close()
+		}
+	}()
+	<-waitForServer
 
 	root := common.HostProc("")
 	v, err := getProcInodesAll(root, 0)
@@ -88,7 +107,7 @@ func TestGetProcInodesAll(t *testing.T) {
 }
 
 func TestConnectionsMax(t *testing.T) {
-	if os.Getenv("CIRCLECI") == "true" {
+	if os.Getenv("CI") != "" {
 		t.Skip("Skip CI")
 	}
 

--- a/process/process_test.go
+++ b/process/process_test.go
@@ -253,7 +253,10 @@ func Test_Process_Groups(t *testing.T) {
 	if err != nil {
 		t.Errorf("getting groups error %v", err)
 	}
-	if len(v) <= 0 || v[0] < 0 {
+	if len(v) == 0 {
+		t.Skip("Groups is empty")
+	}
+	if v[0] < 0 {
 		t.Errorf("invalid Groups: %v", v)
 	}
 }

